### PR TITLE
Refactoring to use the the 3.x SDK

### LINKED ISSUE #133: Refactor & split bootstrap file into smaller components

### LINKED ISSUE #134: Bump SDK to 3.0.0

### LINKED ISSUE #135: Refactor deprecated methods
Tools::display404Error() -> Controller::getController('PageNotFoundController')->run();

### PR DESCRIPTION
**Pull request by mridang**
Closes #133 
Closes #134
Closes #135

- [x] Double check and fix all Nosto backend code to reference the correct `.nosto_*` selectors
- [ ] Language selector is hidden on nosto account config page if there is no authenticated nosto account
- [x] Move all `PS_` terms to be constants
- [x] Move all `id` strings to constants. Use `grep -Ri "'id'" classes controllers nostotagging.php`
- [x] Remove useless `Context` usage. Use `grep -Ri "Context::getContext()->" classes controllers nostotagging.php`
- [ ] Check that all URL building uses the `NostoHelperLink`. If not, why not?
- [ ] Check why we try to find the 800px image, we should always use the original.
- [ ] Alias all SDK imports as `NostoSDK*`

### CODE REVIEW

File: bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        require_once($moduleDir . '/controllers/admin/AdminNostoPersonalizationController.php');
+        require_once($moduleDir . '/controllers/admin/NostoAdvancedSettingController.php');
+        require_once($moduleDir . '/controllers/admin/NostoConnectAccountController.php');
+        require_once($moduleDir . '/controllers/admin/NostoCreateAccountController.php');
+        require_once($moduleDir . '/controllers/admin/NostoUpdateExchangeRateController.php');
+        require_once($moduleDir . '/controllers/admin/NostoIndexController.php');
+        require_once($moduleDir . '/controllers/front/OauthTraitAdapter.php');
+        require_once($moduleDir . '/classes/models/NostoNotification.php');
+        require_once($moduleDir . '/classes/models/NostoExchangeRates.php');
+        require_once($moduleDir . '/classes/helpers/account.php');
```
Comment (mridang) [inline bootstrap.php]: Wrong path.

File: bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        require_once($moduleDir . '/classes/helpers/NostoHelperLogger.php');
+        require_once($moduleDir . '/classes/managers/NostoNotificationManager.php');
+        require_once($moduleDir . '/classes/managers/NostoHookManager.php');
+        require_once($moduleDir . '/classes/services/AbstractNostoService.php');
+        require_once($moduleDir . '/classes/services/NostoProductService.php');
+        require_once($moduleDir . '/classes/services/NostoOrderService.php');
+        require_once($moduleDir . '/classes/services/NostoRatesService.php');
+        require_once($moduleDir . '/classes/services/NostoSettingsService.php');
+        require_once($moduleDir . '/classes/services/NostoSignupService.php');
+        require_once($moduleDir . '/classes/helpers/url.php');
```
Comment (mridang) [inline bootstrap.php]: Some wrong paths here too.

File: classes/blocks/NostoHeaderContent.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * @author    Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2013-2017 Nosto Solutions Ltd
+ * @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+use Nosto\Nosto as NostoSDK;
+
+class NostoHeaderContent
+{
+    const DEFAULT_SERVER_ADDRESS = 'connect.nosto.com';
```
Comment (mridang) [inline classes/blocks/NostoHeaderContent.php]: Should be moved to the SDK

File: bootstrap.php (hunk 1 above)
Comment (chenliangde) [reply bootstrap.php]: Fixed

File: bootstrap.php (hunk 2 above)
Comment (chenliangde) [reply bootstrap.php]: Fixed

**Pull request merged**